commons/src/main/java/io/aiven/kafka/tieredstorage/commons/UniversalRemoteStorageManager.java

```diff
@@ -207,11 +203,34 @@ public void copyLogSegmentData(final RemoteLogSegmentMetadata remoteLogSegmentMe
             }
             uploadIndexFile(remoteLogSegmentMetadata, new ByteBufferInputStream(logSegmentData.leaderEpochIndex()),
                 LEADER_EPOCH);
-        } catch (final StorageBackEndException | IOException e) {
+        } catch (final StorageBackEndException | IOException | InvalidRecordBatchException e) {
             throw new RemoteStorageException(e);
         }
     }
 
+    boolean requiresCompression(final LogSegmentData logSegmentData) throws InvalidRecordBatchException {
```

Comment:
We should think what to do in case of `InvalidRecordBatchException`. Maybe the least disruptive would be to log the error and upload without compression.

Reply:
Great catch. Totally agree, log and leave uncompressed sounds good to me.